modules/processors/nih_dataset_processor.py

```diff
@@ -0,0 +1,59 @@
+import datasets
+import os
+from ..dataset_processor import *
+
+class NIHDataset(Processor):
+
+    def __init__(self, config_name='number',is_query=True, *args, **kwargs):
+        self.dataset_name='nih_v1_'+config_name
+        if is_query:
+            self.dataset_name+='_query'
+        else:
+            self.dataset_name+='_doc'
+        self.conf_name= config_name
+        self.is_query=is_query
+        
+        super().__init__(*args, **kwargs, dataset_name=self.dataset_name)
+    
+    def process(self):
+        print(self.conf_name)
+        d = datasets.load_dataset('naver/bergen_nih_v1',self.conf_name)[self.split]
+        #oracle run data generations
+        qids_dids_l =[ (x['qid'],x['did']) for x in d]
+        #FIXME get the run dirs?
```

Comment:
Here locate the runs relative to this script position, otherwise will fail if the script is run from elsewhere:

Reply:
This is meant to be launched from the main dir. The other options is to add as argument specifying the 'run_dir'. Within the data_processor class, I dont know of how I could get the run dir variable. Maybe from an hydra trick?

Reply:
So overall oracles should not be handled via trec, but via the dataset itself (added as TODO for later, it's quite some change)
There is a way with Hydra but it seems overkill for this usage:

```
from hydra.core.global_hydra import GlobalHydra
from omegaconf import OmegaConf

# Ensure Hydra has been initialized (only required if called in a non-Hydra-managed context)
if not GlobalHydra().is_initialized():
    raise RuntimeError("Hydra has not been initialized. Make sure this script is run within a Hydra context.")

# Access the Hydra configuration
config = GlobalHydra().get_state().cfg
# Now you can use `config` to access any variable in the Hydra config
print(config.some_variable)       # Access a specific variable
```

 which is primarily designed for us. Still I think it matters for this part of the code to work even when the run_dir is not bergen/, so just get the path with something like:
```
# Get the path to the current script
current_script_path = os.path.abspath(__file__)

# Navigate up two levels to get the parent of the parent folder
bergen_run_path = os.path.join(os.path.dirname(os.path.dirname(current_script_path)), 'runs')
```

NB: using this way to access the config would potentially save a lot of arguments passing in bergen!

Reply:
This was not working. I added the trec oracle files in the run dir so need anymore to generate them